src/Databases/DatabaseReplicated.cpp

```diff
@@ -338,42 +338,40 @@ ClusterPtr DatabaseReplicated::getClusterImpl(bool all_groups) const
     return std::make_shared<Cluster>(getContext()->getSettingsRef(), shards, params);
 }
 
-std::vector<UInt8> DatabaseReplicated::tryGetAreReplicasActive(const ClusterPtr & cluster_) const
+ReplicasInfo DatabaseReplicated::tryGetReplicasInfo(const ClusterPtr & cluster_) const
 {
-    Strings paths;
+    ReplicasInfo res;
+
+    auto zookeeper = getZooKeeper();
     const auto & addresses_with_failover = cluster_->getShardsAddresses();
     const auto & shards_info = cluster_->getShardsInfo();
-    for (size_t shard_index = 0; shard_index < shards_info.size(); ++shard_index)
-    {
-        for (const auto & replica : addresses_with_failover[shard_index])
-        {
-            String full_name = getFullReplicaName(replica.database_shard_name, replica.database_replica_name);
-            paths.emplace_back(fs::path(zookeeper_path) / "replicas" / full_name / "active");
-        }
-    }
 
     try
     {
-        auto current_zookeeper = getZooKeeper();
-        auto res = current_zookeeper->exists(paths);
-
-        std::vector<UInt8> statuses;
-        statuses.resize(paths.size());
-
-        for (size_t i = 0; i < res.size(); ++i)
-            if (res[i].error == Coordination::Error::ZOK)
-                statuses[i] = 1;
+        UInt32 max_log_ptr = parse<UInt32>(zookeeper->get(zookeeper_path + "/max_log_ptr"));
 
-        return statuses;
-    }
-    catch (...)
+        for (size_t shard_index = 0; shard_index < shards_info.size(); ++shard_index)
+        {
+            for (const auto & replica : addresses_with_failover[shard_index])
+            {
+                String full_name = getFullReplicaName(replica.database_shard_name, replica.database_replica_name);
+                UInt32 log_ptr = parse<UInt32>(zookeeper->get(fs::path(zookeeper_path) / "replicas" / full_name / "log_ptr"));
+                bool is_active = zookeeper->exists(fs::path(zookeeper_path) / "replicas" / full_name / "active");
```

Comment:
It will work, but it will drastically slow down selects from system.clusters (especially for clusters having a lot of replicas). For each replica, it makes 2 sequential zk requests, so it's 2 * num_of_replicas RTT for each `tryGetReplicasInfo` call. Let's put all paths in a vector, make one [mutli-get request](https://github.com/ClickHouse/ClickHouse/blob/8c4c2b64c5aadeda927b9c2ba2a69e9123fb24b8/src/Common/ZooKeeper/ZooKeeper.h#L332) (`exists` is basically a `get` request + a check for the error code (ZOK/ZNONODE)), and process the list of responses/error codes